packages/frontend/src/components/Explorer.tsx

```diff
@@ -103,10 +104,18 @@ export const Explorer: FC<Props> = ({ savedQueryUuid }) => {
     const [sqlIsOpen, setSqlIsOpen] = useState<boolean>(false);
     const [vizIsOpen, setVizisOpen] = useState<boolean>(!!savedQueryUuid);
     const totalActiveFilters: number = countTotalFilterRules(filters);
+    const [chartId, setChartId] = useState<string>('');
```

Comment:
Do we need this separate state?

What happens if you pass `savedQueryUuid` directly to `useDuplicateMutation`?

Reply:
we just want to be absolutely sure that the `savedQueryUuid` is available before passing it to `useDuplicateMutation` because it can also be undefined, which is why i'm checking and setting it on the useEffect further down.

I have updated this line with `const [chartId] = useState<string>(savedQueryUuid || '');` so we can avoid the useEffect